app/models/reservation.rb

```diff
@@ -110,44 +118,26 @@ def self.unique_equipment_items?(reservations)
     item_ids == item_ids.uniq
   end
 
-  def self.number_for_model_on_date(date, model_id, source)
-    # count the number of reservations that overlaps a date within
-    # a given array of source reservations and that matches
-    # a specific model id
-    number_for(date, model_id, source, :equipment_model_id)
-  end
-
-  def self.number_for_category_on_date(date, category_id, reservations)
-    number_for(date, category_id, reservations, :category_id)
-  end
-
-  def self.number_for(date, value, source, property)
+  # Counts the number of reservations in source that overlap with the given date
+  # If no date is given, defaults to today
+  # If properties are given, only counts the reservation if it has the 
+  # specified attributes
+  def self.number_for(source, date: Time.zone.today, **properties)
     count = 0
     source.each do |r|
```

Comment:
Can we just replace this `each` call with a `reduce` / `inject` call? That should return the overall value so it will clean the method up a little bit.